### DISCUSSION TITLE
Success with PM07SK.NSA and some questions

### DISCUSSION THREAD

**Discussion by w-marco** [JanM321/esphome-lg-controller#63] [General]
So first of all, thank you for your work. Together with the mini board, a 3D printed case and some 3D printed plug which takes Dupont to plug into the board, I was able to control both of my PM07SK.NSA just fine. It is connected to a Dual-Split Outdoor Unit. (I added my wall unit to the Wiki). 

However there's some issues/questions I have:

1. I read that horizontal positioning isn't supported by the controller, that's why it's lacking and only available via remote. A bummer, but I am fine with it once it's set via Remote. I guess there's no way to force enable it and find out?

2. My unit supports the Air Purifier, but the Switch in HA isn't showing up. My YAML contains this line:
 ```
   purifier:
      name: Air Purifier
      id: air_purifier
      icon: mdi:pine-tree

```
but for some reason it won't show the toggle for it, even though it is supported by the unit. Can I somehow fix that?

3. The toggle for jet mode isn't there. You mentioned at some point that one can add a switch by themselves, any pointers on where to start so I can try out?

**Comment by JanM321** [comment]
> (I added my wall unit to the Wiki).

Hm I don't see PM07SK.NSA in there?

> 1. I read that horizontal positioning isn't supported by the controller, that's why it's lacking and only available via remote. A bummer, but I am fine with it once it's set via Remote. I guess there's no way to force enable it and find out?

No, unfortunately it's just not available in the wired controller protocol :/ The LG controllers I have reverse engineered also have no support for this.

> 2. My unit supports the Air Purifier, but the Switch in HA isn't showing up.

You could try to delete [this line](https://github.com/JanM321/esphome-lg-controller/blob/9964e0ad8316ac971dcfcd284e475792635b8043/esphome/components/lg_controller/lg-controller.h#L398) and see if it works then.  The unit is reporting that it doesn't support the purifier setting so it's possible it will just ignore this setting.

> 3. The toggle for jet mode isn't there. You mentioned at some point that one can add a switch by themselves, any pointers on where to start so I can try out?

It should be possible to support Jet mode for cooling, but I don't think it's super easy. You could try something along the lines of the purifier switch and then send fan mode = Power to the unit if set (see protocol.md for the fan mode values).

**Reply by w-marco** [reply]
> 
> Hm I don't see PM07SK.NSA in there?
> 

Well yeah, I forgot :D added now!

> 
> You could try to delete [this line](https://github.com/JanM321/esphome-lg-controller/blob/9964e0ad8316ac971dcfcd284e475792635b8043/esphome/components/lg_controller/lg-controller.h#L398) and see if it works then. The unit is reporting that it doesn't support the purifier setting so it's possible it will just ignore this setting.
> 
This works to add the button, thanks!. Do you have any idea how to test if it does activate the feature? 
There's no symbol on the unit itself for that so I'd need to somehow pull the data to check if it works.

> 
> It should be possible to support Jet mode for cooling, but I don't think it's super easy. You could try something along the lines of the purifier switch and then send fan mode = Power to the unit if set (see protocol.md for the fan mode values).
>
I'll need to have a deeper look into that, thanks, will try when I have some time to spare.

Btw. there's quite some delay (say 5-10 secs) from sending a command to the ESP (e.g. turn on) to actually turning the AC on. Is this due to LGs protocol or is something wrong with my ESPHome config?

**Reply by JanM321** [reply]
> Btw. there's quite some delay (say 5-10 secs) from sending a command to the ESP (e.g. turn on) to actually turning the AC on. Is this due to LGs protocol or is something wrong with my ESPHome config?

Partly due to the slow LG protocol (104 bps so sending a message will take at least a second and more than that if the AC is sending one first) but also internally the controller uses a 6 second 'interval'. So every 6 seconds it will wake up and process/send messages.

**Reply by w-marco** [reply]
> 
> You could try to delete [this line](https://github.com/JanM321/esphome-lg-controller/blob/9964e0ad8316ac971dcfcd284e475792635b8043/esphome/components/lg_controller/lg-controller.h#L398) and see if it works then. The unit is reporting that it doesn't support the purifier setting so it's possible it will just ignore this setting.

At least when I turn off/on the feature using the remote, the status in ESPHome also updates to reflect that. So I would guess setting it via the ESP also should work? 
If it's correctly reporting Purifier status over the Bus I would guess it has general support for it in there so toggling it over the Bus should be working too?

**Reply by JanM321** [reply]
I expect it's working fine then, but hard to say for sure.
